src/components/fields/location-field/location-helper.ts

```diff
@@ -103,41 +103,67 @@ export namespace LocationHelper {
 		}
 	};
 
+	// Added useFallback arg to handle pin location address.
 	export const fetchAddress = async (
 		query: string,
 		pageNumber: number,
+		lat: number,
+		lng: number,
 		onSuccess?: (results: IResultsMetaData) => void,
 		onFail?: (error: unknown) => void
 	) => {
 		if (!query) return;
 		try {
-			const { results, pageNum, totalNumPages } = await searchByAddress({
-				searchValue: query,
-				getAddressDetails: OneMapBoolean.YES,
-				returnGeom: OneMapBoolean.YES,
-				pageNum: pageNumber,
-			});
+			const regex = /^pin location/;
+			if (regex.test(query.toLowerCase())) {
+				const [queryLat, queryLng] = query
+					.split(":")[1]
+					.split(",")
+					.map((value) => parseFloat(value.trim()));
+				let parsedResult = [];
+				if (lat && lng && lat === queryLat && lng === queryLng) {
```

Comment:
would there ever be a scenario that the formvalue.lat and formavalue.lng differs from the values taken from query?

asking bcos I'm wondering why we need to pass in lat lng to `fetchAddress`

Reply:
It was done with the intention of comparing the `lat` and `lng` values in the `query` (for "Pin location: ...") with the `lat` and `lng` values from `formValue`.

In my view, it only impacts queries that contain "Pin location" in its value.

Reply:
There can be a scenario where the `lat`, `lng` values taken from `formValues` differ from the values taken from `query`.

In such cases, the result list will not be populated with the result of a user-modified "Pin location: lat, lng" string, but will fallback on the usual location search flow, which would produce the "No results found..." text.

Reply:
may I know what are the scenarios where the latlng is different from the query? just trying to understand how are we to deal with such scenarios. with this logic, it will yield no results and the user will be confused. I am under the impression we can search based on the query provided?

in addition, for the checking of the query, instead of checking for the presence of `pin location`, manually lower casing it to test and making undefined checks, may I suggest to use the same regex from location-search for it:
`^(pin location:) -?\d{0,3}.\d*, -?\d{0,3}.\d*$`

- this will help check for the presence of `pin location`, exact query format and the presence of the lat and lng
- there won't be the need to check if `lat` or `lng` is undefined; `queryLat` and `queryLng` is definitely defined with the regex check, so the equality checks will fail if `lat` or `lng` is undefined

Reply:
how do we end up in this scenario?

> There can be a scenario where the lat, lng values taken from formValues differ from the values taken from query.

Reply:
Upon revisiting the location modal after navigating back from review screen, we have the `formValues.address`, `formValues.lat`, `formValues.lng` as well as the `queryString` from the prior confirmation.

The scenario where the values can differ is when the user attempts to alter the `queryString` to produce a different pin location value.